widget/tabcontainer.go

```diff
@@ -580,3 +695,150 @@ func (r *tabButtonRenderer) padding() fyne.Size {
 }
+
+func (r *tabButtonRenderer) updateLabel() {
+	r.label.Text = r.button.Text
+	r.label.Color = theme.TextColor()
+	r.label.TextSize = theme.TextSize()
+}
+
+var _ fyne.Widget = (*tabOverflowButton)(nil)
+var _ fyne.Tappable = (*tabOverflowButton)(nil)
+var _ desktop.Hoverable = (*tabOverflowButton)(nil)
+
+type tabOverflowButton struct {
+	tabButton
+	container *TabContainer
+	popUp     *PopUpMenu
+}
+
+func (b *tabOverflowButton) CreateRenderer() fyne.WidgetRenderer {
+	b.ExtendBaseWidget(b)
+	b.updateContent()
+	b.updateOverflowPopUp()
+	return b.tabButton.CreateRenderer()
+}
+
+// Hide hides the tabOverflowButton.
+// Implements: fyne.Widget
+func (b *tabOverflowButton) Hide() {
+	b.ExtendBaseWidget(b)
+	b.hideOverflowPopUp()
+	b.BaseWidget.Hide()
+}
+
+// MinSize returns the size that this widget should not shrink below
+// Implements: fyne.Widget
+func (b *tabOverflowButton) MinSize() fyne.Size {
+	b.ExtendBaseWidget(b)
+	return b.BaseWidget.MinSize()
+}
+
+// Move changes the relative position of the tabOverflowButton.
+// Implements: fyne.Widget
+func (b *tabOverflowButton) Move(pos fyne.Position) {
+	b.ExtendBaseWidget(b)
+	b.BaseWidget.Move(pos)
+	b.Refresh()
+}
+
+// Refresh updates the Icon, Text, and PopUp based on TabContainer.tabLocation.
+// Implements: fyne.Widget
+func (b *tabOverflowButton) Refresh() {
+	b.ExtendBaseWidget(b)
+	b.updateContent()
+	b.updateOverflowPopUp()
+	b.BaseWidget.Refresh()
+}
+
+func (b *tabOverflowButton) createOverflowPopUp() {
+	log.Println("Creating Overflow PopUp")
+	b.container.propertyLock.RLock()
+	tabs := b.container.Items
+	b.container.propertyLock.RUnlock()
+	var items []*fyne.MenuItem
+	for i, item := range tabs {
+		index := i // capture
+		text := item.Text
+		if text == "" {
+			text = fmt.Sprintf("Tab %d", i+1)
+		}
+		items = append(items, fyne.NewMenuItem(text, func() {
+			b.container.propertyLock.Lock()
+			b.container.overflowExpanded = false
+			b.container.propertyLock.Unlock()
+			b.container.SelectTabIndex(index)
+		}))
+	}
+
+	c := fyne.CurrentApp().Driver().CanvasForObject(b.super())
+	b.popUp = newPopUpMenu(fyne.NewMenu("", items...), c)
+	b.popUp.Show()
+}
+
+func (b *tabOverflowButton) hideOverflowPopUp() {
+	if b.popUp != nil {
+		b.popUp.Dismiss()
+		b.popUp.Hide()
+		b.popUp = nil
+	}
+}
+
+func (b *tabOverflowButton) updateContent() {
+	hasText, hasIcon := b.container.contentTypes()
+	if hasText {
+		b.Text = "More"
+	} else {
+		b.Text = ""
+	}
+	if hasIcon {
+		switch b.container.TabLocation() {
+		case TabLocationLeading, TabLocationTrailing:
+			b.Icon = theme.MoreVerticalIcon()
+		default:
+			b.Icon = theme.MoreHorizontalIcon()
+		}
+	} else {
+		b.Icon = nil
+	}
+}
+
+func (b *tabOverflowButton) updateOverflowPopUp() {
+	b.container.propertyLock.RLock()
+	expanded := b.container.overflowExpanded
+	b.container.propertyLock.RUnlock()
+	if expanded {
+		log.Println("Overflow Expanded")
+		if b.popUp == nil {
```

Comment:
It seems like expanded might not be needed - the presence of the popup should be sufficient to know if it should be managed. The tap event of the button could be used to create the overflow, removing it's code from this refresh method. The overflow would be removed by it's dismissal or item selection so that should not have to be handled here either.

Reply:
The tabOverflowButton could manage the popup without the expanded flag, however it would not be possible to programmatically toggle the overflow popup without getting the renderer for the TabContainer to access the tabOverflowButton and tap it. Since the tests need to toggle the overflow they would need to be in the same package, or tabContainerRenderer, tabOverflowButton would need to be exported

Reply:
If this is an important path then another approach to support it would be for the TabContainer to be told which button is the more one, so it can be tapped.
i.e. `contaner.moreButton := more` in the CreateRenderer, or wherever is appropriate.